packages/core/src/modules/connections/services/ConnectionService.ts

```diff
@@ -567,6 +580,10 @@ export class ConnectionService {
       return connection.id === connectionId && connection.state === ConnectionState.Complete
     }
 
+    // Check if already done
+    const connection = await this.connectionRepository.findById(connectionId)
+    if (connection && isConnected(connection)) return connection //TODO: check if this leaves trailing listeners behind?
```

Comment:
I think this could introduce a realllllly small gap where we retrieve the record, and before the listener is started, the connection state is updated to done.

By starting the listener before doing findById, this could never happen. Is there a specific reason you changed the order?

Reply:
I think we could clean this up a bit in the future, but for now I think it works fine